.github/workflows/operate-ci-build-reusable.yml

```diff
@@ -53,11 +61,10 @@ jobs:
       # Setup: define env variables
       - name: Set GitHub environment variables
         run: |
-          GIT_COMMIT_HASH=$(git rev-parse ${{ inputs.branch }})
```

Comment:
Can you explain why this change is necessary and `inputs.commitSha` needs to be introduced?

Reply:
In my understanding the `{{ github.sha }}` and  `git rev-parse branch` deliver different commit hashes. As example for instance in the last commit of this workflow I have set all workflows to use `github.sha`. However, the operate workflow uses rev parse and this results in different tags being generated (you can inspect for instance the last images pushed to [dev harbor](https://dev.registry.camunda.cloud/harbor/projects/91/repositories))

Reply:
I was under the impression that this was a mistake in the implementation, but what is your view on this? Do you know the intention behind the `git rev-parse branch` command in this case? For the preview environmnets we definitely need the `{{ github.sha }}` as we need the hash of the latest commit.

Reply:
Another thing that I noticed is that optimize actually tags its images with [github.event.pull_request.head_sha ](https://github.com/camunda/camunda/blob/bdc52631ca2ae1421619902e03662ab015c3ec1b/.github/workflows/tasklist-ci.yml#L33) but operate and tasklist use `git rev-parse branch` in their ci. Should these be in line? This is not directly connected to my changes but it is maybe important in order to get the logic of the reusable workflows right

Reply:
> In my understanding the {{ github.sha }} and git rev-parse branch deliver different commit hashes.

Which is the reason why I noticed this change.

> For the preview environmnets we definitely need the {{ github.sha }} as we need the hash of the latest commit.

According to [GH docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#github-context) the `github.sha` is:

> The commit SHA that triggered the workflow. The value of this commit SHA depends on the event that triggered the workflow. For more information, see "[Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)."

So on [pull request events](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request) it is

> Last merge commit on the GITHUB_REF branch

Which is not the latest commit on the branch but instead an automatically by GH created merge commit that includes the changes from `main`.

This is the reason why `git rev-parse branch` or `github.event.pull_request.head.sha` are used. [GH docs](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request) also mention:

> Note that GITHUB_SHA for this event is the last merge commit of the pull request merge branch. If you want to get the commit ID for the last commit to the head branch of the pull request, use github.event.pull_request.head.sha instead.

Reply:
> Should these be in line?

Yes they could if the context is correct - the reusable*.yml workflows accept a `branch` input so they probably cannot use `github.event.pull_request.head.sha` since they don't know from which context they will be called.

Reply:
sounds good!